ros2action/test/fixtures/fibonacci_action_introspection.py

```diff
@@ -61,15 +61,26 @@ def __init__(self):
         self._action_client = ActionClient(self, Fibonacci, 'fibonacci')
         self._action_client.configure_introspection(
             self.get_clock(), qos_profile_system_default, ServiceIntrospectionState.CONTENTS)
-        self._timer = self.create_timer(2, self._timer_callback)
+        self._timer = self.create_timer(3, self._timer_callback)
+        self._send_goal_future = None
 
     def _timer_callback(self):
-        self._timer.cancel()
         if not self._action_client.wait_for_server():
-            self.get_logger().info('Action erver is unavailable.')
+            self.get_logger().info('Action server is unavailable.')
             return
 
-        self.send_goal(2)
+        if self._send_goal_future is None:
+            self.send_goal(2)
+            return
+
+        if not self._send_goal_future.done():
+            return
+
+        if self._send_goal_future.result() is None:
+            self.get_logger().error('Exception generated on the goal: {0}'.format(
+                self._send_goal_future.exception()))
+
+        self._send_goal_future = None
```

Comment:
this changes the test fixture behavior a bit. before this is just a one shot timer to start the action communication, now it tries to call the action request if the previous goal is completed. this makes the test process more robust, and it can check the expected output messages are in the command line output with given test time.